src/ShowingTransitionSupport.js

```diff
@@ -0,0 +1,221 @@
+/**
+* This module exports the {@link module:enyo/ShowingTransitionSupport~ShowingTransitionSupport} mixin.
+*
+* @wip
+* @module enyo/ShowingTransitionSupport
+*/
+
+var kind = require('enyo/kind'),
+	utils = require('enyo/utils');
+
+/**
+* The {@link module:enyo/ShowingTransitionSupport~ShowingTransitionSupport} [mixin]{@glossary mixin}
+* is applicable to any control that should use a transition or animation when it is shown or hidden.
+* This mixin adds optional state-classes to the component at its resting or transitioning states.
+* The states are as follows:
+*
+* * 'shown' and 'hidden' - resting, static, not-transitioning, past tense states.
+* * 'showing' and 'hiding' - transitioning-to, progressive tense states.
+*
+* A CSS class may be optionally supplied and applied to the component during that state. It will be
+* removed immediately when the component is no longer in that state. The 'hidden', 'hiding', and
+* 'showing' CSS classes are already defined as defaults, to account for the most common use case.
+* The same classes are allowed to be used on multiple states.
+*
+* It may be desirable to have a transition on only one of the showing or hiding states; this is
+* also possible.
+*
+* Transitions take time, so be sure to include a `hidingDuration` and/or `showingDuration`. You may
+* take advantage of this [mixin]{@glossary mixin}'s state classes without using transitions by
+* leaving the duration properties blank (0, null or undefined) and the resting states will simply be
+* applied immediately, skipping the transition state classes.
+*
+* An optional method may be supplied to fire at the end of either of the transitions, using
+* `hidingMethod` and/or `showingMethod`.
+*
+* @mixin
+* @wip
+* @public
+*/
+var ShowingTransitionSupport = {
```

Comment:
Ack, my fault, @webOS101 says we need this for JSDoc purposes - I'll add this back in.